src/moonlink/src/error.rs

```diff
@@ -1,109 +1,144 @@
 use arrow::error::ArrowError;
 use iceberg::Error as IcebergError;
 use parquet::errors::ParquetError;
+use std::fmt;
 use std::io;
 use std::result;
-use std::sync::Arc;
 use thiserror::Error;
 use tokio::sync::watch;
 
+/// Error status indicating whether an error is retryable
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum ErrorStatus {
+    Permanent,
+    Temporary,
+    Persistent,
+}
+
+/// Custom error struct for moonlink
+#[derive(Clone, Debug)]
+pub struct ErrorStruct {
+    pub message: String,
+    pub status: ErrorStatus,
+    // TODO: take out source now to deal with other field first, should add it back
+}
+
+impl fmt::Display for ErrorStruct {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.message)
```

Comment:
curious 
- `ErrorStruct` is already `Debug`, why do we need another formatting function? Apologize for my poor rust knowledge :sweat_smile:
- why don't we put `status` in the formatting message? My general principle is to put much information in the error message as possible, as long as it's not too spammy.

Reply:
Thank you for the explanation!